src/phase_change/AllenCahnHierarchyIntegrator.cpp

```diff
@@ -0,0 +1,1824 @@
+                // lf_cui_conservative_convective_op->applyConvectiveOperator(
+                //     lf_scratch_idx, H_scratch_idx, lf_N_scratch_idx);
+            }
+            else if (convective_time_stepping_type == TRAPEZOIDAL_RULE)
+            {
+                const int u_new_idx = var_db->mapVariableAndContextToIndex(d_u_adv_var, getNewContext());
+                d_lf_convective_op->setAdvectionVelocity(u_new_idx);
+                d_hier_cc_data_ops->copyData(lf_scratch_idx, lf_new_idx);
+                // H_pre_idx is used to use the same flux used in the advection of H.
+                d_hier_cc_data_ops->copyData(H_scratch_idx, d_H_pre_idx);
+                d_lf_convective_op->setSolutionTime(new_time);
+
+                // Checking whether the CellConvectiveOperator can be used for div (H lf u)
+                d_lf_convective_op->interpolateToFaceOnHierarchy(
+                    d_lf_interp_idx, lf_scratch_idx, u_new_idx, /*synch_cf_bdry*/ false);
+                d_lf_convective_op->interpolateToFaceOnHierarchy(
+                    d_H_interp_idx, H_scratch_idx, u_new_idx, /*synch_cf_bdry*/ false);
+                d_hier_fc_data_ops->multiply(d_lf_flux_idx, d_lf_interp_idx, d_H_interp_idx);
+                d_hier_fc_data_ops->multiply(d_lf_flux_idx, d_lf_flux_idx, u_new_idx);
+                d_lf_convective_op->computeConservativeDerivativeOnHierarchy(
+                    lf_N_scratch_idx, d_lf_flux_idx, /*synch_cf_bdry*/ true);
+
+                // Pointer<AdvDiffCUIConservativeConvectiveOperator> lf_cui_conservative_convective_op =
+                //     d_lf_convective_op;
+                // lf_cui_conservative_convective_op->applyConvectiveOperator(
+                //     lf_scratch_idx, H_scratch_idx, lf_N_scratch_idx);
+            }
+        }
+        if (convective_time_stepping_type == ADAMS_BASHFORTH)
+        {
+#if !defined(NDEBUG)
+            TBOX_ASSERT(cycle_num == 0);
+#endif
+            const int lf_N_old_current_idx = var_db->mapVariableAndContextToIndex(d_lf_N_old_var, getCurrentContext());
+            const double omega = dt / d_dt_previous[0];
+            d_hier_cc_data_ops->linearSum(
+                lf_N_scratch_idx, 1.0 + 0.5 * omega, lf_N_scratch_idx, -0.5 * omega, lf_N_old_current_idx);
+        }
+
+        if (convective_time_stepping_type == ADAMS_BASHFORTH || convective_time_stepping_type == MIDPOINT_RULE)
+        {
+            d_hier_cc_data_ops->axpy(lf_rhs_scratch_idx, -1.0, lf_N_scratch_idx, lf_rhs_scratch_idx);
+        }
+        else if (convective_time_stepping_type == TRAPEZOIDAL_RULE)
+        {
+            d_hier_cc_data_ops->axpy(lf_rhs_scratch_idx, -0.5, lf_N_scratch_idx, lf_rhs_scratch_idx);
+        }
+    }
+
+    // Account for forcing terms.
+    const int lf_F_scratch_idx = var_db->mapVariableAndContextToIndex(d_lf_F_var, getScratchContext());
+    const int lf_F_new_idx = var_db->mapVariableAndContextToIndex(d_lf_F_var, getNewContext());
```

Comment:
Here it is - we should store those patch data indices instead of looking them up just to match what's done elsewhere in IBAMR.